Do not build as a nobody user and don't chown the build directory, no files should be owned by nobody (Closes: #554168)
--- a/dkms
+++ b/dkms
@@ -1529,13 +1529,6 @@ function do_build()
     echo $"`date`" >> "$dkms_tree/$module/$module_version/build/make.log"
     local the_make_command=`echo $make_command | sed "s/^make/make KERNELRELEASE=${kernelver_array[0]}/"`
 
-    #if we're root, try to run as a user instead
-    if [ "$USER" = "root" ] && getent passwd nobody 1>/dev/null && su nobody -c "/bin/true" 1>/dev/null; then
-        the_make_command="su nobody -c \"$the_make_command\""
-        chmod +x $dkms_tree/$module/$module_version/build
-        chown -R nobody $dkms_tree/$module/$module_version/build
-    fi
-
     invoke_command "$the_make_command >> $dkms_tree/$module/$module_version/build/make.log 2>&1" "$the_make_command" background
 
     # Make sure good return status
